packages/core/src/components/button/buttons.tsx

```diff
@@ -73,13 +87,30 @@ export const AnchorButtonFactory = React.createFactory(AnchorButton);
 function getButtonClasses(props: IButtonProps) {
     return classNames(
         Classes.BUTTON,
-        { [Classes.DISABLED]: props.disabled },
+        { [Classes.DISABLED]: isButtonDisabled(props) },
+        { [Classes.LOADING]: props.loading },
         Classes.iconClass(props.iconName),
         Classes.intentClass(props.intent),
         props.className,
     );
 }
 
+function isButtonDisabled(props: IButtonProps) {
+    return props.disabled || props.loading;
+}
+
+function maybeRenderSpinner(loading: boolean) {
+    return loading
+      ? <Spinner className="pt-small pt-button-spinner" />
```

Comment:
Apologies if I asked already, but instead of `.pt-button-spinner`, can't we use `.pt-button.pt-loading .pt-spinner`?